docs/custom_validators.md

```diff
@@ -104,6 +103,87 @@ class CheckInterfaceIPv4(JmesPathModelValidation):  # pylint: disable=too-few-pu
     error = "All core interfaces do not have IPv4 addresses"
 ```
 
+## PydanticValidation
+
+Schema Enforcer supports utilizing Pydantic models for validation. Pydantic models can be loaded two different ways.
+
+1. Store your models in your `validator_directory`.
+2. Load from a separate library using the `schema_enforcer.schemas.PydanticManager`. These must be defined within the `schema_enforcer` configuration file.
+   1. `pydantic_validators` requires a list of library paths to a `PydanticManager` instance.
+
+Both methods will replace the Pydantic `BaseModel` with the `PydanticValidation` class that provides the required `validate` method that uses the `model_validate` method to validate data. The model is set to the original Pydantic model to validate data against.
+
+```python
+class PydanticValidation(BaseValidation):
+    """Basic wrapper for Pydantic models to be used as validators."""
+
+    model: BaseModel
+
+    def validate(self, data: dict, strict: bool = False):
+        """Validate data against Pydantic model.
+
+        Args:
+            data (dict): variables to be validated by validator
+            strict (bool): true when --strict cli option is used to request strict validation (if provided)
+
+        Returns:
+            None
+
+        Use add_validation_error and add_validation_pass to report results.
+        """
+        try:
+            self.model.model_validate(data, strict=strict)
+            self.add_validation_pass()
+        except ValidationError as err:
+            self.add_validation_error(str(err))
+```
+
+### Storing in Validators Directory
+
+The Pydantic models can be located in any Python file within this directory. The only requirement is these are valid Pydantic `BaseModel` subclasses.
+
+These will be loaded and can be referenced by their class name. For example, `Hostname` will show up as `Hostname`.
+
+```yaml
+# jsonschema: Hostname
+---
+hostname: "az-phx-rtr01"
+```
+
+### Load from External Library
+
+As an example, we will look at models that are within our `my_custom_pydantic_models.manager`. If a **prefix** is defined, you can reference the validators like `f"{prefix}/{model.__name__}`.
+
+```python
+"""Load our models to be used for Schema Enforcer."""
+from pydantic import BaseModel
+from schema_enforcer.schemas.manager import PydanticManager
+
+
+class Hostname(BaseModel):
+    hostname: str = Field(pattern="^[a-z]{2}-[a-z]{3}-[a-z]{1,2}[0-9]{2}$")
+
+
+# Prefix is optional and will default to a blank string, aka no prefix.
+# Models is required to pass in custom Pydantic models.
+manager = PydanticManager(prefix="custom", models=[Hostname])
+```
+
+```toml
+[tool.schema_enforcer]
+pydantic_validators = [
+  "my_custom_pydantic_models.manager"
+]
+```
```

Comment:
What python code would need to be implemented in the "Storing in Validators Directory" case above? I assume it'd look similar but not identical to this?

Reply:
For models stored in the ` validators_directory` will be loaded the same way, but nothing is special required from a user. If they want to add some Pydantic models into the validators directory, they can do so without any additional effort from them.

Is that what you were asking?

Reply:
I think I was just looking for some sample code for that case, similar to the sample code here, as a way to compare/contrast the two approaches.

Reply:
Added a small code example above the **YAML** definition in the documentation